bin/run-circleci-tests

```diff
@@ -4,6 +4,18 @@ set -e
 
 make install_system_dependencies
 
+# Ensure Redis is up and running before starting to run any tests:
+# In the past, individual tests from our unit test suite
+# would sometimes fail with a "ConnectionRefusedError"
+# caused by Redis not starting for the container running the tests.
+# The solution that we use below comes from the following support thread:
+# https://discuss.circleci.com/t/redis-server-problems-unable-to-connect-localhost-6379/4040
+until bash -c "sudo service redis-server status"; do
```

Comment:
I'd say we should limit total time waiting for redis start, in case its ermanently disabled, or starting after a long time. The failure mode I see is: 

- Redis starts after 15 minutes (due to something unrelated)
- Test run timeouts on an unrelated change.